tools/docker/tests/test_flows.sh

```diff
@@ -0,0 +1,106 @@
+#find prplmesh folder
+scriptdir="$(cd "${0%/*}"; pwd)"
+installdir="${scriptdir%/*/*/*/*}/build/install"
+
+dbg() {
+    [ "$VERBOSE" = "true" ] && echo "$@"
+}
+
+status() {
+	printf '\033[1;35m'"$@\n"'\033[0m'
+}
+
+err() {
+    printf '\033[1;31m'"$@\n"'\033[0m'
+}
+
+success() {
+    printf '\033[1;32m'"$@\n"'\033[0m'
+}
+
+run() {
+    dbg "$*"
+    "$@" || exit $?
+}
+
+report() {
+    msg="$1"; shift
+    if "$@"; then
+        success "OK $msg"
+    else
+        err "FAIL $msg"
+        error=1
+    fi
+}
+
+start_gw_repeater() {
+    dbg "delete running containers before starting test"
+    docker rm -f $(docker ps -a -q) >/dev/null 2>&1
+    dbg "start gw and repeater"
+    ${scriptdir}/test_gw_repeater.sh >/dev/null 2>&1
+}
+
+send_topology_query() {
+    dbg "Sending toplogy query to al_mac $1"
+    docker exec -it gateway ${installdir}/bin/beerocks_cli -c "bml_trigger_topology_discovery $1" >/dev/null 2>&1
+}
+
+check_log_for_query() {
+    dbg "Confirming topology query was received"
+    docker exec -it repeater cat tmp/$USER/beerocks/logs/beerocks_agent.log | grep -i -q "Topology Query"
+}
+
+test_topology() {
+    status "test topology query"
+    send_topology_query $1
+    check_log_for_query
+}
+
+test_init() {
+    status "test initialization"
+    start_gw_repeater || {
+        err "start GW+Repeater failed, abort"
+        exit 1
+    }
+    AL_MAC=$(docker exec -it gateway ${installdir}/bin/beerocks_cli -c bml_conn_map | grep IRE_BRIDGE | awk '{print $5}' | cut -d ',' -f 1)
+}
+
+usage() {
+    echo "usage: $(basename $0) [-hva]"
+    echo "  options:"
+    echo "      -h|--help - show this help menu"
+    echo "      -v|--verbose - verbosity on"
+    echo "      -a|--all - test all flows"
+    echo "      --test-topology - test the topology discovery flow"
```

Comment:
I like this, having the option to test all or just one.

However, there probably will be many flows after some time. So maybe it is better to have positional arguments for the flows to run, so we don't have to pass it to getopt. And we should then of course use the name of the function `test_topology` so we don't have to match the string either.

Reply:
I'm failing to understand the difference with having long options for running specific tests compared to having them as positional arguments - can you elaborate?

The use case we had in mind is to always use the --all option while developing, and see all of the flows which are not yet implemented as FAILURES.
Then for each new flow we will add the test and it will pass.

In travis, we will only use short options, so every time a new flow is ready, part of the PR is to update `.travis.yaml` to include that flow's test.

Finally - passing no positional arguments == `running test_gw_repeater --rm`

Reply:
Positional arguments -> call it like this:
```
test_flows.sh -v test-topology test-foo test-bar
```
i.e. without the `--`.

In code, it becomes completely different of course. You wouldn't have variables for the different tests, but just something like:
```
for test; do
    ${test}
done
```
(without an `in`, the for loop iterates over the positional arguments).

For handling the `-a` option, you'd do something like
```
set $all_tests
```
where `all_tests` is a variable containing the names of all tests. Obviously.

Reply:
Oh I get it. makes the code much simpler.